### PR TITLE
metric: add some metric by database

### LINKED ISSUE #37892: Adding schema level information to tidb_server_handle_query_duration_seconds_sum
According to https://docs.pingcap.com/tidb/dev/performance-tuning-overview, TiDB already supported the metric `tidb_server_handle_query_duration_seconds_sum` to monitor processing time of queries from applications to TiDB.

But when we shut down all our applications, it means there is no application query to TiDB, the metric still up and down in our dashboard. So it may turn out to be a mix of processing time for both application queries and TiDB internal queries.

Screen Shot 2022-09-16 at 5 27 57 PM

It should support other labels (beside: `instance`, `job`, `sql_type`) so we can filter between the sources of queries, for instance, may be `schema`, `table`.

PTAL @xzhangxian1008 @xiongjiwei .

### PR DESCRIPTION
**Pull request by srstack**
### What problem does this PR solve?

Issue Number: close #37892

Problem Summary: add  some metric by database 

### What is changed and how it works?
This pr is based on a modification of https://github.com/pingcap/tidb/pull/38261
（We need to add a switch to control whether the db label is added to the metric or not.）

Reasons for not adding table labels: https://prometheus.io/docs/practices/naming/#labels

Add additional information about database name to the current metric 
tidb_server_handle_query_duration_seconds
image

tidb_executor_statement_total
image

tidb_server_execute_error_total
image

**Added a switch to control whether to add database labels**

```
[status]
record-db-label = true // default false
```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [x] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: metrics/session.go [hunk 1]
```diff
@@ -207,6 +207,7 @@ const (
 	LblRollback       = "rollback"
 	LblType           = "type"
 	LblDb             = "db"
+	LblTable          = "table"
```
Comment (jackysp) [inline metrics/session.go:210]: No use anywhere?

Comment (srstack) [inline metrics/session.go:210]:
```suggestion
```

**Review by jackysp — APPROVED** [review]
LGTM

File: executor/compiler.go [hunk 2]
```diff
@@ -207,40 +193,21 @@ func CountStmtNode(stmtNode ast.StmtNode, inRestrictedSQL bool) {
...
-	}
-
-	if !config.GetGlobalConfig().Status.RecordQPSbyDB {
-		return
-	}
 
-	dbLabels := getStmtDbLabel(stmtNode)
-	for dbLabel := range dbLabels {
-		metrics.DbStmtNodeCounter.WithLabelValues(dbLabel, typeLabel).Inc()
+	if config.GetGlobalConfig().Status.RecordQPSbyDB || config.GetGlobalConfig().Status.RecordDBLabel {
```
Comment (bb7133) [inline executor/compiler.go:197]: Why do we need both `RecordQPSbyDB` and `RecordDBLabel`, it's confusing IMO.

Comment (srstack) [reply executor/compiler.go:197]: > 

cc https://github.com/pingcap/tidb/pull/9151

Comment (bb7133) [reply executor/compiler.go:197]: I would prefer to deprecate `RecordQPSbyDB` and remove `DbStmtNodeCounter`. Users could use `RecordDBLabel` and `DbStmtNodeCounter` instead.

But we don't have to make it in this PR.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**